google-cloud-spanner/src/main/java/com/google/cloud/spanner/SpannerExceptionFactory.java

```diff
@@ -213,13 +216,31 @@ private static ResourceInfo extractResourceInfo(Throwable cause) {
     return null;
   }
 
+  private static ErrorInfo extractErrorInfo(Throwable cause) {
+    if (cause != null) {
+      Metadata trailers = Status.trailersFromThrowable(cause);
+      if (trailers != null) {
+        return trailers.get(KEY_ERROR_INFO);
+      }
+    }
+    return null;
+  }
+
   static SpannerException newSpannerExceptionPreformatted(
       ErrorCode code, @Nullable String message, @Nullable Throwable cause) {
     // This is the one place in the codebase that is allowed to call constructors directly.
     DoNotConstructDirectly token = DoNotConstructDirectly.ALLOWED;
     switch (code) {
       case ABORTED:
         return new AbortedException(token, message, cause);
+      case RESOURCE_EXHAUSTED:
+        ErrorInfo info = extractErrorInfo(cause);
+        if (info != null
+            && info.getMetadataMap().containsKey("quota_limit")
+            && "AdminMethodQuotaPerMinutePerProject"
```

Comment:
nit: could we extract constants for the literal Strings?

Reply:
Done